### DISCUSSION TITLE
Load Balancing issue with HTTP source/sink and Vector source/sink

### DISCUSSION THREAD

**Discussion by LinTechSo** [vectordotdev/vector#19409] [Q&A] [answered]
Hi
####  Description:
I am currently facing an issue with load balancing when using two vectors that connect to each other with HTTP as the source and sink in a Kubernetes environment. Specifically, when I increase the replica count of the HTTP server vector, the traffic does not seem to be evenly distributed among the replicas. I have checked for keep alive configuration and other related settings but haven't found a solution.
I would appreciate any guidance or assistance in figuring out how to achieve effective load balancing for the HTTP traffic when increasing the replica count in a Kubernetes environment.

#### Description:
Another challenge I'm facing is that when I use the vector type as both the source and sink for these two vectors, I seem to lose the basic authentication feature.
I would like to understand if there are any updates or improvements planned or any open issue for the addition of basic authentication features when using Vector as both the source and sink.

**Comment by LinTechSo** [comment]
From what I gather, it appears that deploying a load balancer at the front end might be a  only solution. Is it correct to assume that using HAProxy for Vector type and Nginx for HTTP type is a viable strategy?

**Comment by dsmith3197** [comment]
How many client Vector instances do you have? If you only have a single client, it's possible that it is directing all of its traffic to a single server Vector instance even though load balancing is properly set up. Besides that, how did you install Vector in Kubernetes? Did you create a service and are you pointing at the service's endpoint in the client Vector? Doing so should properly set up load balancing without introducing HAProxy or Nginx.

**Reply by LinTechSo** [reply]
Hi, thanks for the reply.
 I deployed a Deployment vector using a simple deployment manifest , and both client and server replicas increased to 10.
 The client sends traffic to the vector server via Kubernetes Service: client--> svc --> server
By monitoring resources, I found out that the traffic does not distribute evenly among all the instances; some replicas eat more cpu

I deployed Nginx to load balance traffic using a vector server headless service, and traffic balancing has improved. My question is whether, if the Vector HTTP sink opens a long-lived connection with the Vector HTTP source, it means that the Kubernetes Service L3/L4 layer can't balance the traffic. In this case, we need a smart L7 layer load balancer like Nginx or some ingress controller to handle this job. am i correct ? i mean how can i find out the keepalive configuration of vector http sink for example

**Reply by dsmith3197** [reply, marked as answer]
There is an open issue related to this for the Vector sink https://github.com/vectordotdev/vector/issues/10728. I also opened https://github.com/vectordotdev/vector/issues/19457 to track adding this functionality to the server side.
